UI/MapWnd.cpp

```diff
@@ -4119,24 +4118,26 @@ void MapWnd::InitFieldRenderingBuffers() {
     }
     m_field_scanline_circles.createServerBuffer();
 
-    for (auto& [field_texture, buffers] : m_field_vertices) {
-        if (!field_texture)
-            continue;
-
-        // TODO: why the binding here?
-        glBindTexture(GL_TEXTURE_2D, field_texture->OpenGLId());
-        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP);
-        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP);
+    std::size_t max_buffer_size = 0;
 
-        buffers.first.createServerBuffer();
-        buffers.second.createServerBuffer();
+    auto field_texture_exists = [](const auto& field_vertices_pair) { return !!field_vertices_pair.first; };
+    for (auto& field_vertices : { std::ref(m_field_vertices_not_visible), std::ref(m_field_vertices_visible) }) {
```

Comment:
can these be `std::cref` ?

Reply:
no, it's the problem I was referring to earlier, that I get const reference to something I need to change and don't want to do `const_cast`. Specifically `buffer.createServerBuffer();` needs the buffer not to be `const`

Reply:
note this is the place I could use `boost::range::join` because it luckily involves only two ranges (while originally I wanted `std::views::join`) but decided against it (and against adding boost view/adaptor replacement for `std::views::join`)